lib/model/dbConn.js

```diff
@@ -47,10 +47,37 @@ function loadModels() {
  *
  * @this Reference to the dbConn module itself.
  */
-function init(host, db, port, options, callback) {
+function init(host, db, port, username, password, options, callback) {
     /*jshint camelcase:false, validthis:true */
+    var hosts,
+        url,
+        credentials = '';
 
-    defaultDb = mongoose.createConnection(host, db, port, options);
+    if (username && password) {
+        credentials = username + ':' + password + '@';
+    }
+
+    function addPort(item) {
+        return item + ':' + port;
+    }
+
+    function commaConcat(previous, current, currentIndex) {
+        if (currentIndex !== 0) {
+            previous += ',';
+        }
+
+        previous += current;
+
+        return previous;
+    }
+
+    hosts = host.split(',')
+        .map(addPort)
+        .reduce(commaConcat, '');
+
+    url = 'mongodb://' + credentials + hosts + ':' + port + '/' + db;
```

Comment:
I think the `+ ':' + port` is not needed since this way you may end up with a connection URL such as: `mongodb://gtv:hey@mongodb1:12345,mongodb2:12345**:12345**/database`